vendor/k8s.io/kubernetes/test/e2e/storage/utils/utils.go

```diff
@@ -53,9 +54,16 @@ func PodExec(pod *v1.Pod, bashExec string) (string, error) {
 	return framework.RunKubectl("exec", fmt.Sprintf("--namespace=%s", pod.Namespace), pod.Name, "--", "/bin/sh", "-c", bashExec)
 }
 
+// PodExec2 wraps RunKubectl to execute a bash cmd in target pod
+func PodExec2(f *framework.Framework, pod *v1.Pod, bashExec string) (string, error) {
+	stdout, _, err := f.ExecCommandInContainerWithFullOutput(pod.Name, pod.Spec.Containers[0].Name, "/bin/sh", "-c", bashExec)
```

Comment:
This proposed `PodExec[2]` has a conceptual problem: it assumes that the pod runs in the namespace created for `f`, which might not be the case. `f` might not even *have* a namespace.

I think it makes sense to create a `ExecCommandInContainerWithFullOutput` function in `test/e2e/framework/pods` which takes client set and namespace as parameters instead of using the ones in `f`. The functions in `f` can be kept for now, they simply became wrappers for the actual implementation. Later they can be removed, if it turns out that they don't have enough value by themselves.

Once we have those new functions, `PodExec2` just needs a new client set parameter instead of `f`.

Reply:
ExecCommandInContainerWithFullOutput() is mostly just wrapper which calls ExecWithOptions(), which uses f.ClientSet.
Means that we should create variants of both `ExecCommandInContainerWithFullOutput` and `ExecWithOptions` under `test/e2e/framework/pods` ?